code/game/objects/items/weapons/RCD.dm

```diff
@@ -451,59 +463,63 @@
 		if(checkResource(5, user))
 			to_chat(user, "Deconstructing Wall...")
 			playsound(loc, 'sound/machines/click.ogg', 50, 1)
-			if(do_after(user, 40 * toolspeed, target = A))
+			var/obj/effect/temp_visual/rcd_effect/reverse/E = new(get_turf(A))
+			if(do_after(user, 50 * toolspeed, target = A))
 				if(!useResource(5, user))
 					return FALSE
 				playsound(loc, usesound, 50, 1)
 				var/turf/AT = A
 				AT.ChangeTurf(/turf/simulated/floor/plating)
 				return TRUE
+			qdel(E)
```

Comment:
I'm worried that these deletions of the visual effects can not be reached under some conditions, such as onces creating returns above.

Have you verified the effect properly cleans up under fail scenarios?

Reply:
There are a number of them like this, FYI

Reply:
shouldn't be any situation where it wouldn't qdel the animation at least that I can think of.

also yes you can see an example in the video linked in the PR